tests/kv_test/main.cpp

```diff
@@ -73,7 +83,24 @@ int main(int argc, char* argv[]) {
     size_t guess_left = (rmi_guess >= err) ? rmi_guess - err : 0;
     size_t guess_right = rmi_guess + err;
     size_t true_index = data.rank_within(lookup, guess_left, guess_right);
+   
+    if (answer != true_index) {
+	std::cout << "ERROR: Incorrect RMI Index. Expected value: " << answer
+		  << " RMI guess:" << true_index << std::endl; 
+    }
     
+    if (t_idx + 1 == count_milestone) {
+        auto curr_time = std::chrono::high_resolution_clock::now();
+	auto time_elapsed = std::chrono::duration_cast<std::chrono::nanoseconds>(curr_time - start_t).count();
+	std::cout << "t = " << time_elapsed << " ns: " << t_idx + 1 
+		  << " counts, tot " << (time_elapsed) / (t_idx + 1) 
+		  << "/op, seg " << (time_elapsed - last_elapsed) / (t_idx + 1 - last_count_milestone) << "/op" << std::endl;  
+	last_elapsed = time_elapsed;
+        last_count_milestone = count_milestone;	
+	count_milestone = ceil(((double) count_milestone) * freq_mul);  // next milestone to print       
```

Comment:
Possibly merge this reporting with `report_t`? Feel free to extend its arguments.